modules/objdetect/src/qrcode.cpp

```diff
@@ -2359,10 +2418,86 @@ bool QRDecode::versionDefinition()
         }
     }
     version = saturate_cast<uint8_t>((std::min(transition_x, transition_y) - 1) * 0.25 - 1);
+    if (version_tmp >= 7 || version >= 7) {
+        numModules = numModules >= (21+6*4) ? numModules : 21 + (version - 1) * 4;
+        const double moduleSize = no_border_intermediate.rows / numModules;
+
+        Point2d startVersionInfo1 = Point2d((numModules-8.-3.)*moduleSize, 0.);
+        Point2d endVersionInfo1 = Point2d((numModules-8.)*moduleSize, moduleSize*6.);
+
+        Point2d startVersionInfo2 = Point2d(0., (numModules-8.-3.)*moduleSize);
+        Point2d endVersionInfo2 = Point2d(moduleSize*6., (numModules-8.)*moduleSize);
+
+        Mat v1(no_border_intermediate, Rect2d(startVersionInfo1, endVersionInfo1));
+        Mat v2(no_border_intermediate, Rect2d(startVersionInfo2, endVersionInfo2));
+        resize(v1, v1, Size(3, 6), 0., 0., INTER_NEAREST);
+        resize(v2, v2, Size(6, 3), 0., 0., INTER_NEAREST);
+        Mat version1, version2;
+        rotate((255-v1)/255, version1, ROTATE_180), rotate(((255-v2)/255).t(), version2, ROTATE_180);
+        static uint8_t versionCodes[][18] = {
```

Comment:
const. Please add reference to documentation where it was taken.

Reply:
fixed

Reply:
add const and reference to https://www.thonky.com/qr-code-tutorial/format-version-tables